cmake/projects/scelta/hunter.cmake

```diff
@@ -0,0 +1,42 @@
+# !!! DO NOT PLACE HEADER GUARDS HERE !!!
+
+include(hunter_add_version)
+include(hunter_cacheable)
+#include(hunter_cmake_args)
+#include(hunter_configuration_types)
+include(hunter_download)
+include(hunter_pick_scheme)
+
+hunter_add_version(
+    PACKAGE_NAME
+    scelta
+    VERSION
+    0.1.0-a0f4f70
+    URL
+    "https://github.com/Bjoe/scelta/archive/594aed957f4c2867243c08b73af49ef4d4599520.tar.gz"
```

Comment:
Again, I need a repo in cpp-pm before we can merge this here ....

Reply:
https://github.com/cpp-pm/scelta

Reply:
Thanks, done, let us wait if https://travis-ci.org/Bjoe/hunter/builds/642316673 and https://ci.appveyor.com/project/Bjoe/hunter-t1t06/builds/30375527 are green :-).

Reply:
Looks all red?

Reply:
Arg, sorry my fault. Forgot to update `default_version` now it is green

Reply:
https://travis-ci.org/Bjoe/hunter/builds/642734277 and https://ci.appveyor.com/project/Bjoe/hunter-t1t06/builds/30397451

Reply:
Sometime an example say more than thousand words ;-)

Do following ...

Clone my example:

`git clone https://github.com/Bjoe/hunter.git`

Then checkout:

`git checkout step-1`

Build the project .... after all is fine execute the `bte` binary:

`Version: Botan 2.10.0 (unreleased, revision unknown, distribution unspecified)`

So no surprise ... :-)

Keep eyes on following cmake output:
```
HUNTER_INSTALL_ROOT_DIR= ...
botan-2_LIBRARY_DIRS= ...
```
`HUNTER_INSTALL_ROOT_DIR` is the current hunter path 
`botan-2_LIBRARY_DIRS` is the "cache entry" in CMakeCache.txt which `pkg_check_modules` is set.

Now we update the botan version from 2.10.0 to 2.11.0-110af9494 .... checkout out ...

`git checkout step-2`

Now build the project again for example `cmake --build <YOUR_BUILD_PATH>` ...

Hunter will build botan version 2.11.0-110af9494 and you see also following cmake output like:
```
...
-- [hunter] BOTAN_ROOT: /home/developer/.hunter/_Base/36e7f44/f49d2cd/8d03466/Install (ver.: 2.11.0-110af9494)
...
```
So all looks fine, but when you execute `bte` it will still print:

Version: Botan 2.10.0 (unreleased, revision unknown, distribution unspecified)

If you verify the output from `HUNTER_INSTALL_ROOT_DIR` and `botan-2_LIBRARY_DIRS` ... then you will have some output like:
```
-- HUNTER_INSTALL_ROOT_DIR=/home/developer/.hunter/_Base/36e7f44/f49d2cd/8d03466/Install
-- botan-2_LIBRARY_DIRS=/home/developer/.hunter/_Base/36e7f44/f49d2cd/5f3f694/Install/lib
```
You see .... botan-2_LIBRARY_DIRS is still pointing to the "build" before ... this is really bad and nobody will see this!

This happen, because if `botan-2_FOUND` is 1 then `pkg_check_modules` reads only the environemnt from the CMakeCache.txt. This behaivor is like `find_library` see https://cmake.org/cmake/help/v3.16/command/find_library.html?highlight=find_library

You can test my solution ... checkout

`git checkout step-3`

Now you will get the error message (again should be more "proper").

As @hjmallon [comment](https://github.com/cpp-pm/hunter/pull/139#issuecomment-576557959) all the FindX.cmake working somehow with `find_library`, `find_path` or `find_program` ... all they have the same behaivor. 
Hunter also provides `FindX.cmake` with for example `cmake/find/FindAvahi.cmake` line 51 or `cmake/find/FindBoost.cmake` line 326 .... 
Btw. FindBoost.cmake is also used when you add the config module with `find_package(Boost CONFIG REQUIRED)` ! See `cmake/templates/BoostConfig.cmake.in` line 51 `find_package(Boost MODULE REQUIRED ${Boost_FIND_COMPONENTS})` ...  this will fixed in the [use boost config branch](https://github.com/Bjoe/hunter/tree/pr.boost) ..

Anyway, I will verify if there are more "pitfalls" for example with `find_package` ...